WordPress/wordpress.org
# Add the pattern preview

## Problem

When you see the strings to translate in a pattern, you don't have a live preview to be able to see the strings in context.


## Solution

This PR adds the live preview for the patterns, using the [filter and action added in GlotPress](https://github.com/GlotPress/GlotPress/pull/1665).

![image](https://github.com/WordPress/wordpress.org/assets/1667814/19fa12dc-e1f2-4d3e-83ee-b98ee4374a3a)

![image](https://github.com/WordPress/wordpress.org/assets/1667814/38cc173e-d5c6-46eb-9481-1c8185836bec)

**Responsive preview (under 1280 px)**

![image](https://github.com/WordPress/wordpress.org/assets/1667814/150cbdcb-6b59-4a90-869c-46c2b1085b8b)



## Testing Instructions

You need to use [this GlotPress branch](https://github.com/GlotPress/GlotPress/pull/1665) (or the develop branch if this branch has been merged). 

You need to add the `&pattern_preview=true` text in the URL to enable this preview.

To test this PR:
1. Go to the [patterns page](https://translate.wordpress.org/locale/gl/default/patterns/) in any language.
2. Click on the [link of a pattern](https://translate.wordpress.org/projects/patterns/core/gl/default/?filters%5Bterm%5D=https%3A%2F%2Fwordpress.org%2Fpatterns%2Fpattern%2F3-item-listicle%2F&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated_or_rejected_or_changesrequested_or_old&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc) to see the live preview.
3. Click in another projects like [WordPress.org](https://translate.wordpress.org/projects/meta/wordpress-org/gl/default/), a [theme](https://translate.wordpress.org/projects/wp-themes/beauty-salon-lite/gl/default/) or a [plugin](https://translate.wordpress.org/projects/wp-plugins/ultimate-member/dev/gl/default/) to be sure that the preview is only showed in the patterns.
